### DISCUSSION TITLE
How to communicate with Infisical self-hosted via REST

### DISCUSSION THREAD

**Discussion by andreyddk** [Infisical/infisical#1328] [Q&A]
Hi! I was deploy the infisical to k8s and I would like to get secrets by REST api but I see the many of requests which require X-API-Key in docs. For example https://infisical.com/docs/api-reference/endpoints/folders/list
What is this and how can I get X-API-Key?

**Comment by jodanpotasu** [comment]
I struggled with this a few days ago. Turns out you don't need it, the `bearer` itself is enough. 

You can generate it from the endpoint 'login universal`, where you get this token as a return.
https://infisical.com/docs/api-reference/endpoints/universal-auth/login
